src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -1261,4 +1263,17 @@ public static StreamCodec<RegistryFriendlyByteBuf, Ingredient> ingredientStreamC
             }
         );
     }
+
+    public static GatherItemComponentsEvent gatherItemComponents(Item item) {
+        var event = new GatherItemComponentsEvent(item);
+        MinecraftForge.EVENT_BUS.post(event);
+        return event;
+    }
```

Comment:
This should be moved to ForgeEventFactory, where it can be simplified to:
```suggestion
    public static GatherItemComponentsEvent gatherItemComponents(Item item) {
        return fire(new GatherItemComponentsEvent(item));
    }
```

Reply:
Was told by Lex that it should be in ForgeHooks, via https://discord.com/channels/1129059589325852724/1129095235889270844/1235034708266713179

Reply:
I was referring to all the chode you shoved into the vanilla `components()` method to rebuild/merge the maps.
If you're just firing an event it belongs in Factory